platform/kobo/enable-wifi.sh

```diff
@@ -14,43 +14,84 @@ for fd in /proc/"$$"/fd/*; do
+
+    if ! grep -q "^${kmod} " "/proc/modules"; then
+        insmod "${KMOD_PATH}/${kmod}.ko" "${@}"
+        usleep 250000
+    fi
+}
+
 # Power up WiFi chip
-if [ -n "${SKIP_SDIO_PWR_MODULE}" ]; then
-    # 208 is CM_WIFI_CTRL
-    ./luajit frontend/device/kobo/ntx_io.lua 208 1
-else
-    if ! grep -q "^sdio_wifi_pwr" "/proc/modules"; then
-        if [ -e "/drivers/${PLATFORM}/wifi/sdio_wifi_pwr.ko" ]; then
-            # Handle the shitty DVFS switcheroo...
-            if [ -n "${CPUFREQ_DVFS}" ]; then
-                echo "userspace" >"/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"
-                echo "1" >"/sys/devices/platform/mxc_dvfs_core.0/enable"
-            fi
+case "${POWER_TOGGLE}" in
+    "ntx_io")
+        # 208 is CM_WIFI_CTRL
+        ./luajit frontend/device/kobo/ntx_io.lua 208 1
+        ;;
+    "wmt")
+        # NOTE: Unlike earlier platforms, it seems the WiFi modules are only loaded once, and never unloaded.
+        #       So, just make sure they actually are loaded before we go on...
+        insmod_asneeded "wmt_drv"
+        insmod_asneeded "wmt_chrdev_wifi"
+        insmod_asneeded "wmt_cdev_bt"
+        insmod_asneeded "${WIFI_MODULE}"
 
-            insmod "/drivers/${PLATFORM}/wifi/sdio_wifi_pwr.ko"
-        else
-            # Poke the kernel via ioctl on platforms without the dedicated power module...
-            # 208 is CM_WIFI_CTRL
-            ./luajit frontend/device/kobo/ntx_io.lua 208 1
+        # Black magic courtesy of wmt_dbg_func_ctrl @ (out of tree) modules/connectivity/wmt_mt66xx/common_main/linux/wmt_dbg.c
+        # Enable debug commands
+        echo "0xDB9DB9" >/proc/driver/wmt_dbg
+        # Disable the LPBK test
+        echo "7 9 0" >/proc/driver/wmt_dbg
+        # Nickel appears to sleep for ~1s
+        sleep 1
+        echo "0xDB9DB9" >/proc/driver/wmt_dbg
+        # Enable the LPBK test (this'll block for ~1.3s)
+        echo "7 9 1" >/proc/driver/wmt_dbg
+        # Finally, power on the chip
+        echo 1 >/dev/wmtWifi
+        ;;
+    *)
+        if ! grep -q "^sdio_wifi_pwr " "/proc/modules"; then
+            if [ -e "${KMOD_PATH}/sdio_wifi_pwr.ko" ]; then
+                # Handle the shitty DVFS switcheroo...
+                if [ -n "${CPUFREQ_DVFS}" ]; then
+                    echo "userspace" >"/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"
+                    echo "1" >"/sys/devices/platform/mxc_dvfs_core.0/enable"
+                fi
+
+                insmod "${KMOD_PATH}/sdio_wifi_pwr.ko"
+            else
+                # Poke the kernel via ioctl on platforms without the dedicated power module...
+                ./luajit frontend/device/kobo/ntx_io.lua 208 1
+            fi
         fi
-    fi
-fi
+        ;;
+esac
 # Moar sleep!
 usleep 250000
 
 # Load WiFi modules
 # NOTE: Used to be exported in WIFI_MODULE_PATH before FW 4.23
-if ! grep -q "^${WIFI_MODULE}" "/proc/modules"; then
+if ! grep -q "^${WIFI_MODULE} " "/proc/modules"; then
```

Comment:
Was it at risk of actually matching something it shouldn't or just for sanity?

Reply:
Currently, no, but a common prefix used for a series of interdependent modules is not terribly far fetched (and it was nearly the case here, which prompted that change ^^).